### PR TITLE
[#152517955] Add ADR for selection of language/framework for web apps

### PR DESCRIPTION
**Pull request by alext**
## What

Add an ADR saying that we'll use Ruby and Rails for user-facing apps that have a web interface.

## How to review

* Check the ADR makes sense.
* Run the site as described in the README and verify it renders correctly.

## Who can review

Not me.

### CODE REVIEW

**Comment by paroxp** [comment]
Not reviewing it - just expressing my sadness. :(

File: docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Status
+
+Accepted.
+
+# Consequences
+
+We should consider porting some of our existing applications over to Ruby and Rails.
+
+[govuk_template]: https://github.com/alphagov/govuk_template
+[govuk_frontent_toolkit]: https://github.com/alphagov/govuk_frontend_toolkit
```
Comment (timmow) [inline docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: 🌳 

`s/ent/end`

🌲

Comment (paroxp) [reply docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: Nice spot!

Comment (alext) [reply docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: Fixed...

File: docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Context
+
+We are starting to develop a number of user-facing applications with web
+interfaces that need to be styled to look like GOV.UK etc. In order to keep
+things consistent we want to pick a single programming language and framework
+to write these in.
+
+We've previously used [Sinatra](http://sinatrarb.com/) for this, but ran into
+issues with it's default configuration which isn't secure, leading to an XSS
```
Comment (camelpunch) [inline docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: its

File: docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+vulnerability. We therefore want to choose something that comes with secure
+defaults, and makes it easier to avoid these sort of issues.
```
Comment (camelpunch) [inline docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: "sorts", or "this sort of issue"

File: docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md (hunk 2 above)
Comment (alext) [reply docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: Doh!!... -> #166

File: docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md (hunk 3 above)
Comment (alext) [reply docs/architecture_decision_records/ADR022-web_app_language_and_framework_selection.md]: Also fixed in #166

**Pull request merged**